src/classes/RD2_VisualizeScheduleController.cls

```diff
@@ -115,6 +148,35 @@ public with sharing class RD2_VisualizeScheduleController {
         }
     }
 
+    /**
+     * @description Inner Class of Future Donations for a given Recurring Donation
+     */
+    public class Schedule {
+        @AuraEnabled public Integer scheduleNumber;
+        @AuraEnabled public Decimal amount;
+        @AuraEnabled public String paymentMethod;
+        @AuraEnabled public String campaign;
+        @AuraEnabled public Date startDate;
+        @AuraEnabled public Date endDate;
+        @AuraEnabled public String period;
+        @AuraEnabled public Integer frequency;
+        @AuraEnabled public String dayOfMonth;
+        @AuraEnabled public String currencyIsoCode;
+
+        public Schedule(Integer num, npe03__Recurring_Donation__c rd, RD2_ScheduleService.ActiveSchedule schedule) {
+            this.scheduleNumber = num;
+            this.amount = schedule.installmentAmount;
+            this.paymentMethod = schedule.paymentMethod;
+            this.campaign = schedule.campaignName;
+            this.startDate = schedule.startDate;
+            this.endDate = schedule.endDate;
+            this.period = schedule.installmentPeriod;
+            this.frequency = schedule.installmentFrequency;
+            this.dayOfMonth = schedule.dayOfMonth;
+            this.currencyIsoCode = getCurrencyCode(rd);
```

Comment:
I believe you're passing in the rd here just for `getCurrencyCode()`. I recommend modifying that method to use a parameter type of `SObject` so you can pass in any object (RecurringDonation or RecurringDonationSchedule).

Reply:
Instead of building `ActiveSchedule` in `RD2_ScheduleService`, and then incorporating it into the controller `Schedule`, why not build everything in the `RD2_ScheduleService` and just return it to the controller (need to send the JSON string and then parse it in LWC though). Or vice versa, build the `ActiveSchedule` from RD schedules and RD SObjects here?

Reply:
@npsp-reedestockton , @force2b : can we move `getVisualizedSchedules(rd)` from the schedule service into the visualizer controller?

Reply:
@RadSFDeveloper The `getVisualizedSchedules(rd)` is used by the Opportunity Evaluation Service as well as this controller, so I think it should stay in the ScheduleService class.

@npsp-reedestockton The call to `getCurrencyCode()` can change to pass in the schedule object since it also has the CurrencyIsoCode. With this change, the constructor no longer requires the rd parameter.
```suggestion
            this.currencyIsoCode = getCurrencyCode(schedule);
```

Reply:
Don't believe we can get away with eliminating rd parameter since we have moved getVisualizedSchedules back to RD2_ScheduleService.  schedule is no longer an SObject.